src/uct/sm/cma/cma_iface.c

```diff
@@ -23,7 +29,13 @@ static ucs_config_field_t uct_cma_iface_config_table[] = {
 static ucs_status_t uct_cma_iface_get_address(uct_iface_t *tl_iface,
                                               uct_iface_addr_t *addr)
 {
-    *(pid_t*)addr = getpid();
+    pid_t *iface_addr = (void*)addr;
+
+    *iface_addr = getpid() & ~UCT_CMA_IFACE_ADDR_FLAG_PID_NS;
+    if (!ucs_sys_ns_is_root(UCS_SYS_NS_PID)) {
+        *iface_addr                     |= UCT_CMA_IFACE_ADDR_FLAG_PID_NS;
+        *(ucs_sys_ns_t*)(iface_addr + 1) = ucs_sys_get_ns(UCS_SYS_NS_PID);
```

Comment:
can we add struct for extended cma address like for sm base?

Reply:
yep, added struct for extended address